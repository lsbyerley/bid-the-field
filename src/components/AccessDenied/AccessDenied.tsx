import Link from 'next/link';
import { ExclamationCircleIcon } from '@heroicons/react/24/outline';

const AccessDenied = () => {
  return (
    <div className='max-w-2xl pt-8 pb-24 mx-auto sm:pt-16 sm:px-6 lg:max-w-7xl lg:px-8'>
      <div className='shadow-lg alert'>
        <div>
          <ExclamationCircleIcon
            className='flex-shrink-0 w-6 h-6 stroke-info'
            aria-hidden='true'
          />
          <span>Access denied. You must signin to view this page</span>
        </div>
        <div className='flex-none'>
          <Link href='/auth/signin' className='btn btn-sm'>
            Signin
          </Link>
        </div>
      </div>
    </div>
  );
};

export default AccessDenied;
