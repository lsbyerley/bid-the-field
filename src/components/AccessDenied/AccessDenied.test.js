import React from 'react';
import { render, screen } from 'test-utils';
import AccessDenied from './AccessDenied';

const renderAccessDenied = (props) => {
  const utils = render(<AccessDenied {...props} />);
  const home = utils.container.firstChild;
  return { ...utils, home };
};

describe('AccessDenied', () => {
  it('should render and match snapshot', () => {
    const { home } = renderAccessDenied();
    expect(home).toBeTruthy();
    expect(home).toMatchSnapshot();
  });
});
