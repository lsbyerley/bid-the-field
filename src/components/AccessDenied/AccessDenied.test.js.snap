// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessDenied should render and match snapshot 1`] = `
<div
  class="max-w-2xl pt-8 pb-24 mx-auto sm:pt-16 sm:px-6 lg:max-w-7xl lg:px-8"
>
  <div
    class="shadow-lg alert"
  >
    <div>
      <svg
        aria-hidden="true"
        class="flex-shrink-0 w-6 h-6 stroke-info"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 8v4m0 4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <span>
        Access denied. You must signin to view this page
      </span>
    </div>
    <div
      class="flex-none"
    >
      <a
        class="btn btn-sm"
        href="/auth/signin"
      >
        Signin
      </a>
    </div>
  </div>
</div>
`;
